test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+
+#ifndef FASTDDS_STATISTICS
+    // 1. Compilation flag not set
+    eprosima::fastdds::statistics::dds::DomainParticipant statistics_participant;
+    eprosima::fastrtps::types::ReturnCode_t ret = statistics_participant.enable_statistics_datawriter(
+            HISTORY_LATENCY_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+
+    ret = statistics_participant.enable_statistics_datawriter(
+            "INVALID_TOPIC", inconsistent_qos);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+
+    ret = statistics_participant.disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+
+    ret = statistics_participant.disable_statistics_datawriter("INVALID_TOPIC");
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+#else
+    logError(STATISTICS_DOMAINPARTICIPANT_TEST, "This test is going to fail because API is not yet implemented.")
+
+    // 2. Create DomainParticipant and narrow to eprosima::fastdds::statistics::dds::DomainParticipant
+    eprosima::fastdds::dds::DomainParticipant * participant =
+            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            create_participant(0, eprosima::fastdds::dds::PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
+    eprosima::fastdds::statistics::dds::DomainParticipant * statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    ASSERT_NE(statistics_participant, nullptr);
+
+    // 3 & 4 Enable statistics DataWriters and check that the corresponding topic has been created successfully.
+    // Check also that the type has been correctly registered.
+    eprosima::fastdds::dds::TypeSupport history_latency_type(
+            new eprosima::fastdds::statistics::WriterReaderDataPubSubType);
+    eprosima::fastdds::dds::TypeSupport network_latency_type(
+            new eprosima::fastdds::statistics::Locator2LocatorDataPubSubType);
+    eprosima::fastdds::dds::TypeSupport throughput_type(
+            new eprosima::fastdds::statistics::EntityDataPubSubType);
+    eprosima::fastdds::dds::TypeSupport rtps_traffic_type(
+            new eprosima::fastdds::statistics::Entity2LocatorTrafficPubSubType);
+    eprosima::fastdds::dds::TypeSupport count_type(
+            new eprosima::fastdds::statistics::EntityCountPubSubType);
+    eprosima::fastdds::dds::TypeSupport discovery_type(
+            new eprosima::fastdds::statistics::DiscoveryTimePubSubType);
+    eprosima::fastdds::dds::TypeSupport sample_identity_count_type(
+            new eprosima::fastdds::statistics::SampleIdentityCountPubSubType);
+    eprosima::fastdds::dds::TypeSupport physical_data_type(
+            new eprosima::fastdds::statistics::PhysicalDataPubSubType);
+
```

Comment:
I would check here that the types are not registered and the topics do not exist

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3